packages/@lwc/ssr-runtime/src/reflection.ts

```diff
@@ -11,11 +11,27 @@ import {
     create,
     entries,
     hasOwnProperty,
+    htmlPropertyToAttribute,
     isNull,
     toString,
 } from '@lwc/shared';
 
 import type { LightningElement } from './lightning-element';
+import type { Attributes } from './types';
+
+/**
+ * Remove global HTML attributes that correspond to public API properties. If a
+ * component overrides a global HTML attribute we "disable" reflection through
+ * removal.
+ */
+export function deletePublicApiAttrs(attrs: Attributes, publicProps: Array<string>) {
+    publicProps.forEach((publicPropName) => {
+        const publicAttrName = htmlPropertyToAttribute(publicPropName);
+        if (publicAttrName in attrs) {
+            delete attrs[publicAttrName];
+        }
+    });
+}
```

Comment:
It feels a bit janky to mutate the `attrs` object that is passed in to `generateMarkup`. I feel that creating a clone here and re-assigning the `attrs` object makes more sense.

Also this function is small enough that I'm not sure it needs to go into `@lwc/ssr-runtime`. It's only used in one place as well.